libtenzir/builtins/commands/exec.cpp

```diff
@@ -91,6 +92,80 @@ auto add_implicit_source_and_sink(pipeline pipe) -> caf::expected<pipeline> {
   return pipe;
 }
 
+auto format_metric(const metric& metric) {
+  auto result = std::string{};
+  auto it = std::back_inserter(result);
+  const auto locale = std::locale("en_US.UTF-8");
+  constexpr auto indent = std::string_view{"  "};
+  it = fmt::format_to(it, "operator #{} ({})\n", metric.operator_index + 1,
+                      metric.operator_name);
+  it = fmt::format_to(it, "{}elapsed: {}\n", indent, data{metric.time_elapsed});
+  it = fmt::format_to(it, "{}scheduled: {} ({:.2f}%)\n", indent,
+                      data{metric.time_scheduled},
+                      100.0 * static_cast<double>(metric.time_scheduled.count())
+                        / static_cast<double>(metric.time_elapsed.count()));
+  it = fmt::format_to(it, "{}running: {} ({:.2f}%)\n", indent,
+                      data{metric.time_running},
+                      100.0 * static_cast<double>(metric.time_running.count())
+                        / static_cast<double>(metric.time_elapsed.count()));
+  if (metric.inbound_measurement.unit != "void") {
+    it = fmt::format_to(it, "{}inbound:\n", indent);
+    it = fmt::format_to(
+      it, locale, "{}{}{}: {:L} at a rate of {:.2f}/s\n", indent, indent,
+      metric.inbound_measurement.unit, metric.inbound_measurement.num_elements,
+      static_cast<double>(metric.inbound_measurement.num_elements)
+        / std::chrono::duration_cast<
+            std::chrono::duration<double, std::chrono::seconds::period>>(
+            metric.time_elapsed)
+            .count());
+    if (metric.inbound_measurement.unit != "bytes") {
```

Comment:
Why not `operator_type_name<chunk_ptr>()`?  I don't know if that would make things less readable, but OTOH I'm not a fan of magic strings.